### DISCUSSION TITLE
Is there any way to access raw request/response in page.js objects in App Router using Server Components?

### DISCUSSION THREAD

**Discussion by CaptShmo** [vercel/next.js#49829] [App Router]
Hi, I am trying to convert an existing NextJS 12 project (using pages directory routing method) to the NextJS 13 App Router routing method. My application uses a 3rd party library for oauth which requires passing them the raw request and raw body object so it can handle the oauth. I see I can access params and searchParams from page.js but not the actual request/response objects. I looked into using middleware and route.js to solve this problem but it seems like those solutions might not work because (if I understand correctly) middleware running on the edge can't use the full set of nodejs api libraries and a lot of the urls for my application are "set-in-stone" because of 3rd party services consuming our URL's. 

I keep reading this documentation trying to find out if there's a way to convert my application to app router without changing a bunch of url's and completely restructuring the code, but my old code was using getServerSideProps, which had access to request/response objects in the equivalent of what page.js is now. Any chance there's a way to access the full request/response in page.js that I've missed? I would be okay with converting the app to use route.js as long as I can keep my existing outward facing URL's, but it seems that because route.js and page.js cannot exist in the same folder that maybe this is not possible.

If page.js cannot and will not ever be able to access the request/response object, I think that's important to put in the documentation because we could do this previously with getServerSideProps. Also, if that's the case, can route.js return the rendered result of another page.js route? Then I could keep my existing URLs using route.js and just have them render the result of another path. Not ideal but acceptable. 

Thanks!

**Comment by boncz92** [comment]
Did you ever figure this out? @CaptShmo
